microcosm_pubsub/producer.py

```diff
@@ -116,6 +116,27 @@ def __exit__(self, type, value, traceback):
             self.producer.publish_message(media_type, message, topic_arn, opaque_data)
 
 
+class DeferredBatchProducer(DeferredProducer):
+    def __exit__(self, type, value, traceback):
+        if type is not None:
+            return
+
+        messages = [
+            dict(
+                media_type=media_type,
+                message=message,
+                topic_arn=topic_arn,
+                opaque_data=opaque_data,
+            )
+            for media_type, message, topic_arn, opaque_data in self.messages
+        ]
+
+        self.producer.produce(
+            MessageBatchSchema.MEDIA_TYPE,
```

Comment:
One thing that occurs to me is that we might *not* want to allow publishing on the default topic for these messages or then *every* daemon (i.e. for different projects) with the batch message publisher will receive these messages and try to republish them. Thoughts on explicitly not letting this happen vs being careful and making sure we configure a different (service-specific) topic for these messages?